tests/src/main/java/io/kestra/core/junit/extensions/KestraTestExtension.java

```diff
@@ -62,5 +73,49 @@ public void afterTestExecution(ExtensionContext context) throws Exception {
         super.afterTestExecution(context);
 
         TestsUtils.queueConsumersCleanup();
+
+        KestraTest kestraTest = context.getTestClass()
+            .orElseThrow()
+            .getAnnotation(KestraTest.class);
+        if (kestraTest.startRunner()
+            && applicationContext.containsBean(ExecutionRepositoryInterface.class)
+            && applicationContext.containsBean(QueueInterface.class, Qualifiers.byName(QueueFactoryInterface.KILL_NAMED))) {
+            ExecutionRepositoryInterface executionRepository = applicationContext.getBean(ExecutionRepositoryInterface.class);
+            QueueInterface<ExecutionKilled> killQueue = applicationContext.getBean(QueueInterface.class, Qualifiers.byName(QueueFactoryInterface.KILL_NAMED));
+
+            retryingExecutionKill(executionRepository, killQueue, 10);
+
+            TestRunner.testExecutions.get().clear();
+        }
+    }
+
+    private void retryingExecutionKill(ExecutionRepositoryInterface executionRepository, QueueInterface<ExecutionKilled> killQueue, int retriesLeft) throws InterruptedException {
+        try {
+            TestRunner.testExecutions.get().stream()
+                .flatMap(launchedExecution -> executionRepository.findById(launchedExecution.getTenantId(), launchedExecution.getId()).stream())
+                .filter(inRepository -> inRepository.getState().isRunning())
```

Comment:
```suggestion
                .filter(inRepository -> !inRepository.getState().isTerminated())
```

better to also kill paused/queued exec